coalib/files/Fileproxy.py

```diff
@@ -0,0 +1,46 @@
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq("filename")
+class Fileproxy:
+    """
+    The file proxy is a wrapper object to wrap every file. The file proxy is an
+    object that contains properties about the file currently processed (such as
+    the filename). It contains:
+
+        * The filename, this is the starting point for every other attribute.
+        * The content of the file as a string.
+        * The content of the file, line-splitted.
+
+    Note: Other properties can be included later as and when required.
+
+    The equality of the object is checked just based on its filename.
+    """
+
+    def __init__(self, filename):
+        """
+        Constructs a new fileproxy object.
+        :param filename: The name of the file to load.
+        """
+        self.filename = filename
+        with open(self.filename, encoding="utf-8") as filehandle:
+            self.content = filehandle.read()
+        self.lines = tuple(self.content.splitlines(True))
```

Comment:
can we make the lines and content readonly somehow? We had some discussions that if bears try to fiddle with the lines/content that should raise an exception because other bears within the process might get the changes as well and we don't want that. It's important that this one is immutable.